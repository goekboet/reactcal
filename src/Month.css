.Month {
    width: 100%;
    margin: 5px;
}

.Label {
    text-align: center;
    font-size: 4vw;
}

.Label > h1 {
    display: inline-block;
    vertical-align: middle;
    margin-left: 10px;
    margin-right: 10px;
}

.Label > a {
    vertical-align: middle;
}

.Calendar {
    display: table;
    width: 100%;
}

.Header {
    display: table-row;
}

.Weekday {
    display: table-cell;
    text-align: center;
}

.Week {
    display: table-row;
}

div.Week > div:first-child .WeekLabel {
    opacity: 1;
}