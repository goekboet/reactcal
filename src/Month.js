import React from 'react';
import Day from './Day';
import './Month.css';

const Month = (props) =>
{
    return (
    <div className="Month">
        <div className="Label">
        <a>◀</a>
        <h1>{props.label}</h1>
        <a>▶</a>
        </div>
        <div className="Calendar">
            <div className="Header">
                {props.weekdays.map(wd => 
                    <div 
                        key={wd} 
                        className="Weekday">{wd}
                    </div>)}
            </div>
            {props.weeks.map(w => 
            <div key={w.key} className="Week">
                {w.days.map(d => <Day 
                    key={d.key} 
                    belongs={d.belongs} 
                    week={d.week} 
                    date={d.date}/>)}
            </div>
            )}
        </div>
    </div>);
};

Month.propTypes = {
    label: React.PropTypes.string,
    weeks: React.PropTypes.arrayOf(React.PropTypes.object),
    weekdays: React.PropTypes.arrayOf(React.PropTypes.string)
}

export default Month;