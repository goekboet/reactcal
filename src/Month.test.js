import React from 'react';
import ReactDOM from 'react-dom';
import Month from './Month';
import renderer from 'react-test-renderer';
import testdata from './Calendars/Months.testdata';

describe('Month view', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Month label="test" weeks={[]} weekdays={[]} />, div);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(
      <Month 
        label={testdata.label} 
        weeks={testdata.weeks}
        weekview={(a, b) => () => undefined} 
        weekdays={testdata.weekdays} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })
});

