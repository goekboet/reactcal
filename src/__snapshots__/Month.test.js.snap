exports[`Month view matches snapshot 1`] = `
<div
  className="Month">
  <div
    className="Label">
    <a>
      ◀
    </a>
    <h1>
      Mars 2017
    </h1>
    <a>
      ▶
    </a>
  </div>
  <div
    className="Calendar">
    <div
      className="Header">
      <div
        className="Weekday">
        Mon
      </div>
      <div
        className="Weekday">
        Tue
      </div>
      <div
        className="Weekday">
        Wed
      </div>
      <div
        className="Weekday">
        Thu
      </div>
      <div
        className="Weekday">
        Fri
      </div>
      <div
        className="Weekday">
        Sat
      </div>
      <div
        className="Weekday">
        Sun
      </div>
    </div>
    <div
      className="Week">
      <div
        className="Day">
        <p
          className="Date">
          27
        </p>
        <p
          className="WeekLabel">
          v9
        </p>
      </div>
      <div
        className="Day">
        <p
          className="Date">
          28
        </p>
        <p
          className="WeekLabel">
          v9
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          1
        </p>
        <p
          className="WeekLabel">
          v9
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          2
        </p>
        <p
          className="WeekLabel">
          v9
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          3
        </p>
        <p
          className="WeekLabel">
          v9
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          4
        </p>
        <p
          className="WeekLabel">
          v9
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          5
        </p>
        <p
          className="WeekLabel">
          v9
        </p>
      </div>
    </div>
    <div
      className="Week">
      <div
        className="Day Belongs">
        <p
          className="Date">
          6
        </p>
        <p
          className="WeekLabel">
          v10
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          7
        </p>
        <p
          className="WeekLabel">
          v10
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          8
        </p>
        <p
          className="WeekLabel">
          v10
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          9
        </p>
        <p
          className="WeekLabel">
          v10
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          10
        </p>
        <p
          className="WeekLabel">
          v10
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          11
        </p>
        <p
          className="WeekLabel">
          v10
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          12
        </p>
        <p
          className="WeekLabel">
          v10
        </p>
      </div>
    </div>
    <div
      className="Week">
      <div
        className="Day Belongs">
        <p
          className="Date">
          13
        </p>
        <p
          className="WeekLabel">
          v11
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          14
        </p>
        <p
          className="WeekLabel">
          v11
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          15
        </p>
        <p
          className="WeekLabel">
          v11
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          16
        </p>
        <p
          className="WeekLabel">
          v11
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          17
        </p>
        <p
          className="WeekLabel">
          v11
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          18
        </p>
        <p
          className="WeekLabel">
          v11
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          19
        </p>
        <p
          className="WeekLabel">
          v11
        </p>
      </div>
    </div>
    <div
      className="Week">
      <div
        className="Day Belongs">
        <p
          className="Date">
          20
        </p>
        <p
          className="WeekLabel">
          v12
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          21
        </p>
        <p
          className="WeekLabel">
          v12
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          22
        </p>
        <p
          className="WeekLabel">
          v12
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          23
        </p>
        <p
          className="WeekLabel">
          v12
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          24
        </p>
        <p
          className="WeekLabel">
          v12
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          25
        </p>
        <p
          className="WeekLabel">
          v12
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          26
        </p>
        <p
          className="WeekLabel">
          v12
        </p>
      </div>
    </div>
    <div
      className="Week">
      <div
        className="Day Belongs">
        <p
          className="Date">
          27
        </p>
        <p
          className="WeekLabel">
          v13
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          28
        </p>
        <p
          className="WeekLabel">
          v13
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          29
        </p>
        <p
          className="WeekLabel">
          v13
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          30
        </p>
        <p
          className="WeekLabel">
          v13
        </p>
      </div>
      <div
        className="Day Belongs">
        <p
          className="Date">
          31
        </p>
        <p
          className="WeekLabel">
          v13
        </p>
      </div>
      <div
        className="Day">
        <p
          className="Date">
          1
        </p>
        <p
          className="WeekLabel">
          v13
        </p>
      </div>
      <div
        className="Day">
        <p
          className="Date">
          2
        </p>
        <p
          className="WeekLabel">
          v13
        </p>
      </div>
    </div>
  </div>
</div>
`;
